draft-ietf-mls-architecture.md

```diff
@@ -696,6 +696,109 @@ software is updated while the client is a member of a group. Thus, in addition
 to preventing downgrade attacks, the members of a group can also observe when it
 is safe to upgrade to a new ciphersuite or protocol version.
 
+# Interoperability Requirements
+
+A fully functional deployment of MLS requires a number of infrastructure
+services to be available, as well as several policies and design trade-offs that
+are decided by the application. This section lists all of the dependencies of an
+MLS deployment that are external to the protocol specification, but would still
+need to be aligned for two deployments to potentially interoperate.
+
+MLS has the built-in ability to negotiate protocol versions, ciphersuites,
+extensions, and credential types. For two deployments to interoperate, they must
+have overlapping support in each of these categories.
+
+MLS relies on the following network services. These network services would
+either need to be the same, or compatible at the wire-format level, in order for
+two different deployments based on them to interoperate.
+
+- An **Authentication Service**, described fully in {{authentication-service}},
+  defines the types of credentials which may be used in a deployment and
+  provides methods for:
+  1. Issuing new credentials,
+  2. Validating existing credentials against a reference identifier, and
+  3. Validating whether or not two credentials represent the same user.
+- A **Delivery Service**, described fully in {{delivery-service}}, provides
+  methods for:
+  1. Delivering messages sent to a group to all members in the group.
+  2. Delivering Welcome messages to new members of a group.
+  3. Downloading KeyPackages for specific clients, and uploading new KeyPackages
+     for a user's own clients.
+- Additional services may or may not be required depending on the application
+  design:
+  - If assisted joining is desired (meaning that the ratchet tree is not
+    provided in Welcome messages), there must be a method to download the
+    ratchet tree corresponding to a group.
+  - If assisted joining is desired and the Delivery Service is not able to
+    compute the ratchet tree itself (because some proposals or commits are sent
+    encrypted), there must be a method for group members to publish the updated
+    ratchet tree after each commit.
```

Comment:
You could summarize this in one bullet with something like "The tree can be either computed by the DS or uploaded by a group member after each Commit."